solver/src/metrics.rs

```diff
@@ -0,0 +1,107 @@
+use std::{convert::TryInto, time::Instant};
+
+use anyhow::Result;
+use model::order::Order;
+use prometheus::{IntCounter, IntCounterVec, IntGaugeVec, Opts, Registry};
+use strum::{AsStaticRef, VariantNames};
+
+use crate::liquidity::Liquidity;
+
+pub trait SolverMetrics {
+    fn liquidity_fetched(&self, liquidity: &[Liquidity]);
+    fn settlement_computed(&self, solver_type: &str, start: Instant);
+    fn order_settled(&self, order: &Order);
+}
+
+// TODO add labeled interaction counter once we support more than one interaction
+pub struct Metrics {
+    trade_counter: IntCounter,
+    order_settlement_time: IntCounter,
+    solver_computation_time: IntCounterVec,
+    liquidity: IntGaugeVec,
+}
+
+impl Metrics {
+    pub fn new(registry: &Registry) -> Result<Self> {
+        let trade_counter =
+            IntCounter::new("gp_v2_solver_trade_counter", "Number of trades settled")?;
+        registry.register(Box::new(trade_counter.clone()))?;
+
+        let order_settlement_time = IntCounter::new(
+            "gp_v2_solver_order_settlement_time_seconds",
+            "Counter for the number of seconds between creation and settlement of an order",
+        )?;
```

Comment:
Maybe better as a histogram?

Reply:
This and the trade counter could also both be calculated based on the events we already have in the database. Not sure if Grafana can easily visualize that though so it's fine if we have it here too.

Reply:
I was thinking about using a histogram (and agree that long term it's the better visualization), but then thought that given we will only have measurements in the beginning it might be nice to see individual times rather than bucketed ones. I also didn't really know what a good bucketing strategy would be so maybe we make it more lossy as data increases and it becomes more clear what relevant buckets could be?

For aggregated past data (e.g. report what was our average wait time in the last months), maybe querying db events might be better (I don't think it can be easily visualized in Grafana though).